server/src/main/java/org/elasticsearch/common/util/concurrent/PrioritizedThrottledTaskRunner.java

```diff
@@ -77,7 +77,7 @@ protected void pollAndSpawn() {
 
     // Each worker thread that runs a task, first needs to get a "free slot" in order to respect maxRunningTasks.
     private boolean incrementRunningTasks() {
-        int preUpdateValue = runningTasks.getAndUpdate(v -> v < maxRunningTasks ? v + 1 : v);
+        int preUpdateValue = runningTasks.getAndAccumulate(maxRunningTasks, (v, maxRunningTasks) -> v < maxRunningTasks ? v + 1 : v);
```

Comment:
Might want to rename `maxRunningTasks` here to make it clear its a different variable, rather than just relying on hiding